force-app/main/default/lwc/newCandidateWithJobApplication/newCandidateWithJobApplication.js

```diff
@@ -0,0 +1,112 @@
+import { LightningElement } from "lwc";
+import getFormFieldsByObjectType from "@salesforce/apex/CandidateAndJobApplicationModal.getFormFieldsByObjectType";
+import { NavigationMixin } from "lightning/navigation";
+import { showErrorToast, showSuccessToast } from "c/utils";
+import Create_Candidate from "@salesforce/label/c.Create_Candidate";
+import Candidate from "@salesforce/label/c.Candidate";
+import Job_Application from "@salesforce/label/c.Job_Application";
+import Cancel_Label from "@salesforce/label/c.Cancel";
+import Save_Label from "@salesforce/label/c.Save";
+import Job_Application_Saved from "@salesforce/label/c.Job_Application_Saved";
+
+export default class NewCandidateWithJobApplication extends NavigationMixin(
+  LightningElement
+) {
+  labels = {
+    Create_Candidate,
+    Candidate,
+    Job_Application,
+    Cancel_Label,
+    Save_Label,
+    Job_Application_Saved
+  };
+  objectApiNames = {
+    candidate: "Candidate__c",
+    jobApplication: "Job_Application__c"
+  };
+  candidateFields = [];
+  jobApplicationFields = [];
+  candidateId;
+
+  connectedCallback() {
```

Comment:
Fyi LWC supports async-awaits, so both styles can be applied (promises or async-awaits, do not mix them)